vscodejs/index.js

```diff
@@ -805,13 +828,24 @@ module.exports = vscode => {
 
   /** Get the name of the directory that contains the current file, editor, as a string.
    *
-   * For example, within my `~/init.js` pressing `cmd+e` on the following:
+   * For example, within my `~/.init.js` pressing `cmd+e` on the following:
    * ```
    * E.currentDirectory() //  ⇒  /Users/musa/
    * ```
    */
   E.currentDirectory = () => E.currentFileName().split('/').slice(0, -1).join('/')
 
+  /** Expand user directory
+   *  expand "~" to actual path.
+   */
+  E.expanduser = ( pathstr ) => {
+    if( pathstr.indexOf("~") === 0) {
+      return path.normalize(path.join(userDir, pathstr.slice(2)))
```

Comment:
Why is this better than plain old `pathstr.replace(/~/g, process.env.HOME)`?

Could you please adorn the documentation of `expandUser` with some examples.

Reply:
Please consider some extreme case like a path "~/foo/~bar/".
expandUser is a function mimic the python's os.path.expanuser

> os.path.expanduser(path)
> On Unix and Windows, return the argument with an initial component of ~ or ~user replaced by that user’s home directory.
> 
> On Unix, an initial ~ is replaced by the environment variable HOME if it is set; otherwise the current user’s home directory is looked up in the password directory through the built-in module [pwd](https://docs.python.org/3/library/pwd.html#module-pwd). An initial ~user is looked up directly in the password directory.
> 
> On Windows, USERPROFILE will be used if set, otherwise a combination of HOMEPATH and HOMEDRIVE will be used. An initial ~user is handled by checking that the last directory component of the current user’s home directory matches USERNAME, and replacing it if so.
> 
> If the expansion fails or if the path does not begin with a tilde, the path is returned unchanged.
> 



Changed in version 3.8: No longer uses HOME on Windows.